v3/lint/global_configurations.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package lint
+
+// Global is what one would intuitive think of as being the global context of the configuration file.
+// That is, given the following configuration...
+//
+// some_flag = true
+// some_string = "the greatest song in the world"
+//
+// [e_some_lint]
+// some_other_flag = false
+//
+// The fields `some_flag` and `some_string` will be targeted to land into this struct.
+type Global struct{}
+
+func (g Global) namespace() string {
+	return "Global"
+}
+
+// RFC5280Config is the higher scoped configuration which services as the deserialization target for...
+//
+// [RFC5280Config]
+// ...
+// ...
+type RFC5280Config struct{}
```

Comment:
Is the idea for these pre-defined config structs to gain various useful fields as part of this change, or is this PR only going to add the scaffolding while individual configurable fields will be added to these later?

Reply:
The later (if I understood you correctly).

So the idea for these structs is to have a ready-to-go facility for when someone (inevitably) finds a configuration that is valid for a broad classification of lints. What I've done here is, as you say, set up the scaffolding such that when that day does eventually arrive it is quite clear how it would be accomplished.